yellowbrick/utils/nan_warnings.py

```diff
@@ -0,0 +1,49 @@
+"""
+Small helpers that help find missing data.
+"""
+
+import numpy as np
+import warnings
+from yellowbrick.exceptions import DataWarning
+
+
+def warn_if_nans_exist(X):
+    """Warn if nans exist in a numpy array."""
+    null_count = count_rows_with_nans(X)
+    total = len(X)
+    percent = 100 * null_count / total
+
+    if null_count > 0:
+        warning_message = \
+            'Warning! Found {} rows of {} ({:0.2f}%) with nan values. Only ' \
+            'complete rows will be plotted.'.format(null_count, total, percent)
+        warnings.warn(warning_message, DataWarning)
+
+
+def count_rows_with_nans(X):
+    """Count the number of rows in 2D arrays that contain any nan values."""
+    if X.ndim == 2:
+        return np.where(np.isnan(X).sum(axis=1) != 0, 1, 0).sum()
+
+
+def count_nan_elements(data):
+    """Count the number of elements in 1D arrays that are nan values."""
+    if data.ndim == 1:
+        return np.isnan(data).sum()
+
+
+def clean_data(X, y=None):
```

Comment:
I'll be honest, I'm not 100% sure about the name here; when used in combination with the module, nan_warnings.clean_data it seems to make sense; but elsewhere I think it might not be communicative. Perhaps, `filter_missing`, `remove_nan_rows` or `clean_nans` would be better? 

Also, I think this function will get used in a few places, so perhaps it would be good to expand the docstring? Something like:

```python 
def clean_data(X, y=None):
    """
    Removes rows that contain np.nan values in data. If y is given, 
    this function ensures that X and y will be filtered together so that 
    their shape remains identical, e.g. rows in X with nans will also 
    remove rows in y, or rows in y with np.nans will also remove 
    corresponding row in X. 

    Parameters
    ------------
    X : array-like 
        Data in shape (m, n) that possibly contains np.nan values 

    y : array-like, optional 
        Data in shape (m, 1) that possibly contains np.nan values 

    Returns
    --------
    X' : np.array 
       Possibly transformed X with any row containing np.nan removed 

    y' : np.array 
        If y is given, will also return possibly transformed y to match the 
        shape of X'. 

    Notes
    ------ 
    This function will return either a np.array if only X is passed or a tuple 
    if both X and y is passed. Because all return values are indexable, it is
    important to recognize what is being passed to the function to determine
    its output. 
    """
```

I know that's a mouthful - feel free to edit as you see fit. We can then add it to the documentation; perhaps in the next PR?

Reply:
1. Thanks for being honest! I totally agree on the name being unclear. :smile:
> There are only two hard things in Computer Science: cache invalidation and naming things.
> -- Phil Karlton

2. I'll add the nice docstring.

Reply:
Thinking more about this main "switch" function, I might prefer to remove it (and simply make two functions) so that reasoning about different returns is eliminated.